src/main/scala/com/fortysevendeg/android/sqlite/SQLDroidConnection.scala

```diff
@@ -115,9 +119,14 @@ class SQLDroidConnection(
     super.finalize()
   }
 
-  private[this] def withOpenDatabase[T](f: (SQLDroidDatabase) => T) = sqliteDatabase match {
-    case Some(db) => f(db)
-    case None => throw new SQLException(alreadyClosedErrorMessage)
+  def withOpenDatabase[T](f: (SQLDroidDatabase) => T) = sqliteDatabase match {
+    case Some(db) =>
+      Try(f(db)) match {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Default case not specified.](https://www.codacy.com/app/47deg/scala-sqlite-droid/file/811484930/issues/source?bid=2442814&fileBranchId=2740117#l124)